sources/packages/backend/apps/api/src/services/student/student.service.ts

```diff
@@ -503,39 +504,59 @@ export class StudentService extends RecordDataModelService<Student> {
+   * is updated by the ministry.
    * @returns updated student, if some data was changed.
    */
-  async synchronizeFromUserToken(
-    studentToken: StudentUserToken,
+  async updateStudentUserData(
+    studentUserData: StudentUserData,
+    options?: { userId?: number },
   ): Promise<Student> {
-    const studentToSync = await this.getStudentById(studentToken.studentId);
+    const userId = studentUserData.userId
+      ? studentUserData.userId
+      : options?.userId;
+    const studentToSync = await this.getStudentById(studentUserData.studentId);
     let mustSave = false;
-    if (studentToken.givenNames === undefined) {
-      studentToken.givenNames = null;
+    if (studentUserData.givenNames === undefined) {
+      studentUserData.givenNames = null;
     }
     if (
-      !dayjs(studentToken.birthdate).isSame(studentToSync.birthDate) ||
-      studentToken.lastName !== studentToSync.user.lastName ||
-      studentToken.givenNames !== studentToSync.user.firstName
+      !dayjs(studentUserData.birthdate).isSame(studentToSync.birthDate) ||
+      studentUserData.lastName.toLowerCase() !==
+        studentToSync.user.lastName.toLowerCase() ||
+      studentUserData.givenNames.toLowerCase() !==
+        studentToSync.user.firstName.toLowerCase()
     ) {
-      studentToSync.birthDate = studentToken.birthdate;
-      studentToSync.user.lastName = studentToken.lastName;
-      studentToSync.user.firstName = studentToken.givenNames;
+      studentToSync.birthDate = studentUserData.birthdate;
+      studentToSync.user.lastName = studentUserData.lastName;
+      studentToSync.user.firstName = studentUserData.givenNames;
       const sinValidation = new SINValidation();
       sinValidation.student = studentToSync;
       sinValidation.sin = studentToSync.sinValidation.sin;
       studentToSync.sinValidation = sinValidation;
       mustSave = true;
     }
     // This condition is not added above, as email does not trigger SIN validation request.
-    if (studentToken.email !== studentToSync.user.email) {
-      studentToSync.user.email = studentToken.email;
+    if (studentUserData.email !== studentToSync.user.email) {
+      studentToSync.user.email = studentUserData.email;
       mustSave = true;
     }
 
     if (mustSave) {
-      studentToSync.modifier = { id: studentToken.userId } as User;
-      return await this.save(studentToSync);
+      return this.dataSource.transaction(async (transactionalEntityManager) => {
+        await this.noteSharedService.createStudentNote(
+          studentToSync.id,
+          NoteType.General,
+          "Student information updated.",
+          userId,
+          transactionalEntityManager,
+        );
+        studentToSync.modifier = { id: userId } as User;
+        return transactionalEntityManager
+          .getRepository(Student)
+          .save(studentToSync);
+      });
     }
 
     // If information between token and SABC DB is same, then just returning without the database call.
```

Comment:
These comments must be adjusted since the token is no longer the only way to have it updated.

Reply:
I have removed the comment. Seems straightforward to understand without the comment.